crates/weaver_semconv/src/any_value.rs

```diff
@@ -0,0 +1,336 @@
+    String {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+    },
+
+    /// An array of strings attribute.
+    #[serde(rename = "string[]")]
+    Strings {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+    },
+
+    /// An array of integer attribute.
+    #[serde(rename = "int[]")]
+    Ints {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+    },
+
+    /// An array of double attribute.
+    #[serde(rename = "double[]")]
+    Doubles {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+    },
+
+    /// An array of boolean attribute.
+    #[serde(rename = "boolean[]")]
+    Booleans {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+    },
+
+    /// The value type is a map of key, value pairs
+    Map {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+        /// The collection of key, values where the value is an `AnyValueSpec`
+        fields: Vec<AnyValueSpec>,
+    },
+
+    /// The value type will just be a bytes.
+    Bytes {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec
+    },
+
+    /// The value type is not specified.
+    Undefined {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec
+    },
+}
+
+/// The Common Value specification for properties associated with an "AnyValue", this
+/// is similar to the current `AttributeSpec` as at the proto level an Attribute
+/// is defined as a "KeyValue".
+/// While this is (currently) a duplication of the existing AttributeSpec, this is
+/// to reduce the size of the change set.
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq, Hash, JsonSchema)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "snake_case")]
+pub struct AnyValueCommonSpec {
+    /// String that uniquely identifies the enum entry.
+    pub id: String,
+    /// A brief description of the value
+    #[serde(skip_serializing_if = "String::is_empty")]
+    #[serde(default)]
+    pub brief: String,
+    /// A more elaborate description of the value.
+    /// It defaults to an empty string.
+    #[serde(skip_serializing_if = "String::is_empty")]
+    #[serde(default)]
+    pub note: String,
+    /// Specifies the stability of the value.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub stability: Option<Stability>,
+    /// Sequence of examples for the value or single example
+    /// value. If only a single example is provided, it can
+    /// directly be reported without encapsulating it
+    /// into a sequence/dictionary.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub examples: Option<Examples>,
```

Comment:
I'm going to add the requirement_level to this on the "basis" that as `AnyValue` is effectively become what is called the `BodyField` in my current PR, it's "missing" these "Common" definitions in this version of the Spec